src/MSBuild.UnitTests/TerminalLogger_Tests.cs

```diff
@@ -229,6 +237,39 @@ public Task PrintBuildSummary_SucceededWithWarnings()
             return Verify(_outputWriter.ToString(), _settings).UniqueForOSPlatform();
         }
 
+        [Fact]
+        public Task PrintImmediateWarningMessage_Succeeded()
+        {
+            InvokeLoggerCallbacksForSimpleProject(succeeded: true, () =>
+            {
+                WarningRaised?.Invoke(_eventSender, MakeWarningEventArgs("[CredentialProvider]"));
```

Comment:
Can the example string be an exact match for one of the NuGet strings?

I also considered pulling a resource from the NuGet assemblies but that seems very heavyweight.